File: Java/9_Queue/LC_649_Dota2Senate.java
class Solution {
    public String predictPartyVictory(String senate) {
        Deque<Integer> rd = new ArrayDeque<>();
        Deque<Integer> dd = new ArrayDeque<>();
        int n = senate.length();
        for (int i = 0; i < n; i++) {
            if (senate.charAt(i) == 'R') {
                rd.addLast(i);
            } else {
                dd.addLast(i);
            }
        }

        while (rd.size() != 0 && dd.size() != 0) {
            int a = rd.pollFirst();
            int b = dd.pollFirst();
            if (a < b) {
                rd.addLast(a + n);
            } else {
                dd.addLast(b + n);
            }
        }

        return rd.size() != 0 ? "Radiant" : "Dire";
    }
}

public class LC_649_Dota2Senate {

}

File: Swift/Leetcode.playground/Sources/21_Heap_PriorityQueue/LC_649_Dota2Senate.swift
class Solution {
    func predictPartyVictory(_ senate: String) -> String {
        var rd = [Int]()
        var dd = [Int]()
        let n = senate.count
        for (i, char) in senate.enumerated() {
            if char == "R" {
                rd.append(i)
            } else {
                dd.append(i)
            }
        }
        while !rd.isEmpty && !dd.isEmpty {
            let a = rd.removeFirst()
            let b = dd.removeFirst()
            if a < b {
                rd.append(a + n)
            } else {
                dd.append(b + n)
            }
        }
        return !rd.isEmpty ? "Radiant" : "Dire"
    }
}